tests/integration/test_in_code_mechanism.py

```diff
@@ -0,0 +1,233 @@
+from acom_music_box import MusicBox, Conditions
+import mechanism_configuration as mc
+
+import math
+import logging
+logger = logging.getLogger(__name__)
+
+
+class TestInCodeMechanism:
+
+    def define_system(self):
+        """
+        Define the system for the test case.
+        """
+        A = mc.Species(name="A")
+        B = mc.Species(name="B")
+        C = mc.Species(name="C")
+        D = mc.Species(name="D")
+        E = mc.Species(name="E")
+        F = mc.Species(name="F")
+        species = { "A": A, "B": B, "C": C, "D": D, "E": E, "F": F }
+        gas = mc.Phase(name="gas", species=list(species.values()))
+
+        return gas, species
+
+    def create_box_model(self, mechanism):
+        """
+        Create a MusicBox instance and load the mechanism.
+        """
+        box_model = MusicBox()
+        box_model.load_mechanism(mechanism)
+
+        # Set the initial conditions
+        box_model.initial_conditions = Conditions(
+          temperature=298.15,
+          pressure=101325.0,
+          species_concentrations={
+            "A": 1.0,
+            "B": 0.0,
+            "C": 0.0,
+            "D": 10.0,
+            "E": 0.0,
+            "F": 0.0,
+          })
+
+        # Set the evolving conditions
+        box_model.add_evolving_condition(
+          300.0,
+          Conditions(
+            temperature=310.0,
+            pressure=100100.0,
+            species_concentrations={
+              "D": 1.0,
+              "E": 0.0,
+              "F": 0.0,
+            }))
+        box_model.add_evolving_condition(
+          450.0,
+          Conditions(
+            temperature=280.0,
+            pressure=90500.0,
+            species_concentrations={
+              "A": 100.0,
+              "B": 0.0,
+              "C": 0.0,
+            }))
+
+        # Set the time step and duration
+        box_model.box_model_options.simulation_length = 600.0
+        box_model.box_model_options.chem_step_time = 2
+        box_model.box_model_options.output_step_time = 6
```

Comment:
Should we rename these to options? Maybe we should also allow these to additionally be set in the constructor

```suggestion
        box_model.options.simulation_length = 600.0
        box_model.options.chem_step_time = 2
        box_model.options.output_step_time = 6
```